java/src/com/ibm/streamsx/topology/internal/logic/Constants.java

```diff
@@ -14,7 +18,19 @@
     private final List<T> data;
     
     public Constants(List<T> data) {
-        this.data = data;
+        List<T> copiedData;
+        
+        if (data.isEmpty())
+            copiedData = data;
+        if (data instanceof ArrayList || data instanceof LinkedList
+                || data instanceof Stack || data instanceof Vector)
+            copiedData = data;
+        else {
+            copiedData = new ArrayList<>(data.size());
```

Comment:
To ensure the list is serializable, for example a list returned from HashMap.values() is not serializable. I'll add a comment.